ws-nextjs-app/pages/_document.page.tsx

```diff
@@ -111,7 +146,19 @@ export default class AppDocument extends Document<DocProps> {
       originalRenderPage({
         enhanceApp: App => props => (
           <CacheProvider value={cache}>
-            <App {...props} />
+            <App
+              {...props}
+              // These are props passed down to ALL pages
+              pageProps={{
+                ...props.pageProps,
+                ...extractHeaders(ctx.req?.headers || {}),
+                toggles,
+                isApp,
+                isAmp,
+                isLite,
+                isNextJs: true,
```

Comment:
Adding props here means we don't need to define them in each `getServerSideProps` `props` object, so its useful for more 'global' style props that all pages will have.